src_c/transform.c

```diff
@@ -45,30 +45,11 @@ struct _module_state {
     SMOOTHSCALE_FILTER_P filter_expand_Y;
 };
 
-#if defined(SCALE_MMX_SUPPORT)
+#define GETSTATE(m) ((struct _module_state *)PyModule_GetState(m))
 
+#ifdef SCALE_MMX_SUPPORT
 #include <SDL_cpuinfo.h>
-
-#define GETSTATE(m) PY3_GETSTATE(_module_state, m)
-
-#else /* if defined(SCALE_MMX_SUPPORT) */
-
-static void
-filter_shrink_X_ONLYC(Uint8 *, Uint8 *, int, int, int, int, int);
-static void
-filter_shrink_Y_ONLYC(Uint8 *, Uint8 *, int, int, int, int, int);
-static void
-filter_expand_X_ONLYC(Uint8 *, Uint8 *, int, int, int, int, int);
-static void
-filter_expand_Y_ONLYC(Uint8 *, Uint8 *, int, int, int, int, int);
-
-static struct _module_state _state = {
-    "GENERIC", filter_shrink_X_ONLYC, filter_shrink_Y_ONLYC,
-    filter_expand_X_ONLYC, filter_expand_Y_ONLYC};
-#define GETSTATE(m) PY2_GETSTATE(_state)
-#define smoothscale_init(st)
```

Comment:
shouldn't some of this still be around?

Reply:
No, I refactored this file a bit because it was not making proper use of module state storage mechanism. Now it's all fixed up and tested (feel free to do more testing :smile:)

Reply:
I mean, if MMX is not defined, it's supposed to have filter_ methods defined here. And now it doesn't.

Unless I'm missing something else in the bigger picture.

Reply:
> I mean, if MMX is not defined, it's supposed to have filter_ methods defined here. And now it doesn't.

Actually it still does, the onlyc filter implementation is a few lines below, I just removed the forward declaration because it was redundant

Reply:
It's 1000 lines below 😄

But ok